### PR TITLE
BHBC-1577 + BHBC-1589: Updated project search page and backend

### PR DESCRIPTION
**Pull request by NickPhura**
# Overview

## Links to jira tickets

https://quartech.atlassian.net/browse/BHBC-1577
https://quartech.atlassian.net/browse/BHBC-1589

## This PR contains the following changes

- {List all the relevant changes. If there are many changes across many Jira tickets, organize the changes by Jira ticket}

## This PR contains the following types of changes

- [ ] New feature (change which adds functionality)
- [ ] Enhancement (improvements to existing functionality)
- [ ] Bug fix (change which fixes an issue)
- [ ] Misc cleanup / Refactoring / Documentation
- [ ] Version change

## Checklist

A list of items that are good to consider when making any changes.

_Note: this list is not exhaustive, and not all items are always applicable._

### General

- [ ] I have performed a self-review of my own code

### Code

- [ ] New files/classes/functions have appropriately descriptive names and comment blocks to describe their use/behaviour
- [ ] I have avoided duplicating code when possible, moving re-usable pieces into functions
- [ ] I have avoided hard-coding values where possible and moved any re-usable constants to a constants file
- [ ] My code is as flat as possible (avoids deeply nested if/else blocks, promise chains, etc)
- [ ] My code changes account for null/undefined values and handle errors appropriately
- [ ] My code uses types/interfaces to help describe values/parameters/etc, help ensure type safety, and improve readability

### Style

- [ ] My code follows the established style conventions
- [ ] My code uses native material-ui components/icons/conventions when possible

### Documentation

- [ ] I have commented my code sufficiently, such that an unfamiliar developer could understand my code
- [ ] I have added/updated README's and related documentation, as needed

### Tests

- [ ] I have added/updated unit tests for any code I've added/updated
- [ ] I have added/updated the Postman requests/tests to account for any API endpoints I've added/updated

### Linting/Formatting

- [ ] I have run the linter and fixed any issues, as needed  
       _See the `lint` commands in package.json_
- [ ] I have run the formatter and fixed any issues, as needed  
       _See the `format` commands in package.json_

### SonarCloud

- [ ] I have addressed all SonarCloud Bugs, Vulnerabilities, Security Hotspots, and Code Smells

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Screenshots

Please add any relevant UI screenshots, if applicable.

### CODE REVIEW

File: app/src/hooks/api/useProjectApi.ts [hunk 1]
```diff
@@ -91,12 +92,17 @@ const useProjectApi = (axios: AxiosInstance) => {
...
    * @param {IProjectAdvancedFilterRequest} filterFieldData
-   * @return {*}  {Promise<IGetProjectsListResponse[]>}
+   * @return {*}  {Promise<IGetProjectForViewResponse[]>}
    */
   const getProjectsList = async (
     filterFieldData?: IProjectAdvancedFilterRequest
-  ): Promise<IGetProjectsListResponse[]> => {
-    const { data } = await axios.post(`/api/projects`, filterFieldData || {});
+  ): Promise<IGetProjectForViewResponse[]> => {
+    const { data } = await axios.get(`/api/project/list`, {
```
Comment (shreypdev) [inline app/src/hooks/api/useProjectApi.ts:100]: We are fetching list of projects so why not just keep /api/projects/list?

File: app/src/features/projects/list/ProjectsListPage.tsx [hunk 2]
```diff
@@ -225,11 +105,12 @@ const ProjectsListPage: React.FC = () => {
         <Table>
           <TableHead>
             <TableRow>
-              <TableCell>Name</TableCell>
+              <TableCell>Project Name</TableCell>
```
Comment (shreypdev) [inline app/src/features/projects/list/ProjectsListPage.tsx:108]: Why do we need multiple TableHead when we are just rendering things in body?

**Review by shreypdev — COMMENTED** [review]
Overall the code looks great. Really nice work @KjartanE and @NickPhura on query redefine and search functionality. Just have few concerns:

-> Its not responsive, we dont want to go till mobile but with small window size it should look good.

-> Project list table heading seems to be having issue as there is no End Date in header.

-> If I query something and then go to project details. when I hit back button from project details page, the filters are cleared. Should be an easy way to retain filters as it is based on url. Check if this can be fixed. 

->Tests are broken.

I will do review one more time once these issues are fixed.

**Comment by NickPhura** [comment]
@KjartanE Was trying out the chip logic, and something isn't quite right. If I add multiple funding agencies (more than 2), whenever I click to delete one of them, it removes all of them except for 1. (Cant remember if you said you were still working on this, so disregard if its still a WIP).

**Comment by KjartanE** [comment]
> @KjartanE Was trying out the chip logic, and something isn't quite right. If I add multiple funding agencies (more than 2), whenever I click to delete one of them, it removes all of them except for 1. (Cant remember if you said you were still working on this, so disregard if its still a WIP).

Resolved!

**Review by anissa-agahchen — APPROVED** [review]

**Review by shreypdev — APPROVED** [review]
Looks solid 💪🏼. Great work, now it is responsive and code is clean as well.

**Pull request merged**